morphia/src/main/java/org/mongodb/morphia/AdvancedDatastore.java

```diff
@@ -257,51 +257,78 @@
     void setDecoderFact(DBDecoderFactory fact);
 
     /**
-     * Inserts an entity in to the named collection.
+     * Inserts an entity in to the mapped collection.
      *
-     * @param collection the collection to update
-     * @param entity     the entity to insert
-     * @param <T>        the type of the entity
+     * @param entity the entity to insert
+     * @param <T>    the type of the entity
      * @return the new key of the inserted entity
      */
-    <T> Key<T> insert(String collection, T entity);
+    <T> Key<T> insert(T entity);
 
     /**
      * Inserts an entity in to the mapped collection.
      *
      * @param entity the entity to insert
+     * @param wc     the WriteConcern to use when inserting
      * @param <T>    the type of the entity
      * @return the new key of the inserted entity
+     * @see WriteConcern
+     * @deprecated use {@link #insert(Object, InsertOptions)}
      */
-    <T> Key<T> insert(T entity);
+    @Deprecated
+    <T> Key<T> insert(T entity, WriteConcern wc);
 
     /**
      * Inserts an entity in to the mapped collection.
      *
-     * @param entity the entity to insert
-     * @param <T>    the type of the entity
+     * @param entity  the entity to insert
+     * @param options the options to apply to the save operation
+     * @param <T>     the type of the entity
      * @return the new key of the inserted entity
+     * @since 1.3
      */
-    <T> Iterable<Key<T>> insert(Iterable<T> entity);
+    <T> Key<T> insert(T entity, InsertOptions options);
```

Comment:
What does it/should it mean to pass as an argument an InsertOptions instance with a null writeConcern property?  The Java driver, in similar situations, applies the default write concern for the DBCollection.  I suggest that Morphia do the same by always applying DatastoreImpl#getWriteConcern to any operation that passes an options class instance with a null write concern.